cadquery/occ_impl/assembly.py

```diff
@@ -289,3 +301,121 @@ def toJSON(
         rv.extend(toJSON(child, loc, color, tolerance))
 
     return rv
+
+
+def toFusedCAF(
+    assy: AssemblyProtocol,
+    assy_name: str,
+    glue: Optional[bool] = False,
+    tol: Optional[float] = None,
+) -> TDocStd_Document:
+    """
+    Converts the assembly to a fused compound and saves that within the document
+    to be exported in a way that preserves the face colors. Because of the use of
+    boolean operations in this method, performance may be slow in some cases.
+
+    :param assy: Assembly that is being converted to a fused compound for the document.
+    :param assy_name: Label that is applied to the top level object in the document.
+    """
+
+    # Prepare the document
+    app = XCAFApp_Application.GetApplication_s()
+    doc = TDocStd_Document(TCollection_ExtendedString("XmlOcaf"))
+    app.InitDocument(doc)
+
+    # Shape and color tools
+    shape_tool = XCAFDoc_DocumentTool.ShapeTool_s(doc.Main())
+    color_tool = XCAFDoc_DocumentTool.ColorTool_s(doc.Main())
+
+    # To fuse the parts of the assembly together
+    fuse_op = BRepAlgoAPI_Fuse()
+    args = TopTools_ListOfShape()
```

Comment:
You could start with a single list and then split it just before passing to fuse_op. This way quite some logic below could be removed.